services/maven-central/maven-central.service.js

```diff
@@ -56,52 +63,26 @@ module.exports = class MavenCentral extends LegacyService {
     ]
   }
 
-  static registerLegacyRouteHandler({ camp, cache }) {
-    camp.route(
-      /^\/maven-central\/v\/([^/]*)\/([^/]*)(?:\/([^/]*))?\.(svg|png|gif|jpg|json)$/,
-      cache((data, match, sendBadge, request) => {
-        const groupId = match[1] // eg, `com.google.inject`
-        const artifactId = match[2] // eg, `guice`
-        const versionPrefix = match[3] || '' // eg, `1.`
-        const format = match[4] || 'gif' // eg, `svg`
-        // Based on repo1.maven.org rather than search.maven.org because of #846.
-        const metadataUrl = `${'http://repo1.maven.org/maven2' +
-          '/'}${encodeURIComponent(groupId).replace(
-          /\./g,
-          '/'
-        )}/${encodeURIComponent(artifactId)}/maven-metadata.xml`
-        const badgeData = getBadgeData('maven-central', data)
-        request(
-          metadataUrl,
-          { headers: { Accept: 'text/xml' } },
-          (err, res, buffer) => {
-            if (err != null) {
-              badgeData.text[1] = 'inaccessible'
-              sendBadge(format, badgeData)
-              return
-            }
-            xml2js.parseString(buffer.toString(), (err, data) => {
-              if (err != null) {
-                badgeData.text[1] = 'invalid'
-                sendBadge(format, badgeData)
-                return
-              }
-              try {
-                const versions = data.metadata.versioning[0].versions[0].version.reverse()
-                const version = versions.find(
-                  version => version.indexOf(versionPrefix) === 0
-                )
-                badgeData.text[1] = versionText(version)
-                badgeData.colorscheme = versionColor(version)
-                sendBadge(format, badgeData)
-              } catch (e) {
-                badgeData.text[1] = 'invalid'
-                sendBadge(format, badgeData)
-              }
-            })
-          }
-        )
-      })
-    )
+  static get defaultBadgeData() {
+    return { label: 'maven-central' }
+  }
+
+  async fetch({ groupId, artifactId }) {
+    const url = `${'https://repo1.maven.org/maven2/'}${encodeURIComponent(
+      groupId
+    ).replace(/\./g, '/')}/${encodeURIComponent(artifactId)}/maven-metadata.xml`
```

Comment:
I found that doing this as multiple function calls inline in a template string broken over 3 lines makes it quite difficult to read. What do you reckon to something more like:

```js
const group = encodeURIComponent(groupId).replace(/\./g, '/')
const artifact = encodeURIComponent(artifactId)
const url = `https://repo1.maven.org/maven2/${group}/${artifact}/maven-metadata.xml`
```

Its still 3 vertical lines, but easier to understand what's going on.